packages/blockchain-wallet-v4-frontend/src/scenes/CoinPage/components/CoinPage/hooks/useChartBalancePanel/useChartBalancePanel.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useMemo } from 'react'
+
+import { ChartBalancePanel } from '../../../ChartBalancePanel'
+import { useCoinPrice } from '../useCoinPrice'
+import { useCurrency } from '../useCurrency'
+import { useFormatFiat } from '../useFormatFiat'
+import { UseChartBalancePanel } from './types'
+
+export const useChartBalancePanel: UseChartBalancePanel = ({ coin }) => {
+  const currency = useCurrency()
+
+  const coinPrice = useCoinPrice({ coin })
+
+  const currentPrice = coinPrice?.currentPrice
+
+  const formattedCurrentPrice = useFormatFiat({
+    currency,
+    value: currentPrice ?? 0
+  })
+
+  const overallChange: number = useMemo(() => {
+    if (!coinPrice) return 0
+
+    try {
+      return parseFloat(coinPrice.overallChange.diff)
+    } catch (err) {
+      return 0
+    }
+  }, [coinPrice])
+
+  const formattedHourPrice = useFormatFiat({
+    currency,
+    value: overallChange
+  })
+
+  const chartBalancePanelNode = useMemo(() => {
+    if (!coinPrice) {
+      return <ChartBalancePanel coinCode={coin} pastHourChange='--' pastHourPrice='--' price='--' />
```

Comment:
@blockdylanb this is the loading state of the balance panel

should I implement the loading state with the grey boxes like how you did in the StandardRow?